USFMToolsSharp/HtmlRenderer.cs

```diff
@@ -112,8 +112,25 @@ private string RenderMarker(Marker input)
                     break;
                 case CMarker cMarker:
                     output.AppendLine("<div class=\"chapter\">");
-                    output.AppendLine($"<span class=\"chaptermarker\">{cMarker.PublishedChapterMarker}</span>");
-                    foreach(Marker marker in input.Contents)
+
+                    if (cMarker.GetChildMarkers<CLMarker>().Count > 0)
+                    {
+                        output.AppendLine($"<div class=\"chaptermarker\">{cMarker.CustomChapterLabel}</div>");
+                        output.AppendLine("<br/>");
+                    }
+                    else if (currentChapterLabel != null)
+                    {
+                        output.AppendLine($"<div class=\"chaptermarker\">{currentChapterLabel} {cMarker.PublishedChapterMarker}</div>");

```

Comment:
Shouldn't this be a different class then chaptermarker?

Reply:
It is now chapterlabel